conn.go

```diff
@@ -71,6 +71,7 @@ type Conn struct {
 
 	// number of replica acks required when publishing to a partition
 	requiredAcks int32
+	apiVersions  []ApiVersion
```

Comment:
Is there a guarantee on the ordering of the values here? I'm wondering how the lookups will work.

Reply:
I can check for ordering. But I just realized that for our purposes it, probably, makes more sense to put it in a map: map[api key]ApiVersion

Reply:
A map makes sense to me 👍